src/wallet/db.h

```diff
@@ -332,27 +333,32 @@ class BerkeleyBatch
         return (ret == 0);
     }
 
-    Dbc* GetCursor()
+    bool CreateCursor()
     {
         if (!pdb)
-            return nullptr;
-        Dbc* pcursor = nullptr;
-        int ret = pdb->cursor(nullptr, &pcursor, 0);
+            return false;
+        m_cursor = nullptr;
```

Comment:
In commit "walletdb: Handle cursor internally" (2711caa2a2d4f04b848c25b614b353a1a08351b0)

I think this needs to either return false if m_cursor is not null or assert that it is null, otherwise calling this at the wrong time will silently leak memory

Reply:
Done